### PR TITLE
Added api, module, module planner class and update build.gradle

### PR DESCRIPTION
**Pull request by sohwkjames**

### CODE REVIEW

File: src/main/java/seedu/duke/Api.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    String urlString = "https://api.nusmods.com/v2/2020-2021/moduleList.json";
+
+    public Api() {
+    }
+
+    public Module[] getAllModules() throws IOException {
+        URL url = new URL(urlString);
+        URLConnection conn = url.openConnection();
+        InputStream is = conn.getInputStream();
+        BufferedReader bR = new BufferedReader(  new InputStreamReader(is));
```
Comment (IcyNoPeople) [inline src/main/java/seedu/duke/Api.java:28]: The gradle build test for ubuntu detect a error for bR (not sure why). Also for the `new BufferedReader(  new InputStreamReader(is));`, they throw error for having a space between the '(' and "new"

File: src/main/java/seedu/duke/Api.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        String line = "";
+
+        StringBuilder responseStrBuilder = new StringBuilder();
+        while((line =  bR.readLine()) != null){
+
```
Comment (IcyNoPeople) [inline src/main/java/seedu/duke/Api.java:33]: I think they might be throwing errors with this empty line

File: src/main/java/seedu/duke/Api.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    public Module[] getAllModules() throws IOException {
+        URL url = new URL(urlString);
+        URLConnection conn = url.openConnection();
+        InputStream is = conn.getInputStream();
+        BufferedReader bR = new BufferedReader(  new InputStreamReader(is));
+        String line = "";
+
+        StringBuilder responseStrBuilder = new StringBuilder();
+        while((line =  bR.readLine()) != null){
```
Comment (IcyNoPeople) [inline src/main/java/seedu/duke/Api.java:32]: > WhitespaceAround: 'while' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround].

 I think the while need to have a space after it as indicated by the error. But it seem weird to have a space between there. Can try adding a space to see if the error will be solve?

File: src/main/java/seedu/duke/Api.java (hunk 2 above)
Comment (sohwkjames) [reply src/main/java/seedu/duke/Api.java]: resolved

**Pull request merged**